plugin/pkg/cache/cache.go

```diff
@@ -31,10 +31,6 @@ type shard struct {
 // New returns a new cache.
 func New(size int) *Cache {
 	ssize := size / shardSize
-	if ssize < 512 {
-		ssize = 512
-	}
-
```

Comment:
No minimum at all ?
I guess to work properly we should verify that each shard of the cache is at least 1 or 2 size large No ?

Reply:
I think without having the minimum,  0 will has the effect of disable the cache.

Reply:
are you sure?

What are you seeing OOMs or something?

Reply:
Someone having OOM on k8s, did investigation on the memory used by cache : https://github.com/kubernetes/kops/issues/5652

Looks like we are using an average of 850 bytes for each entry of the cache.
and because we do not clean it when TTL are done (to avoid extra locks on the shards), then the cache keep growing up to 260MB. with other usage of memory, it is then recommended to have the memory allocated to 320MB **if you have only one cache**.
Things are getting worst if you use several stanza with cache on each of them ...

I guess that having ability to define small cache can help a lot (especially for stub domains).

With this change of code, we are now back to 10k entries as defaults (instead of 131k). Which means each cache, if you keep default value would use about 10Mb, and you could tune smaller cache for the stub domains.

Reply:
i don't think we want to totally remove this check, but lowering is fine.

We also bluntly cache entire messages which is also inefficient memory, but easy to reason about.
And note, note: duing the cache refactor this was an oversight on my part, that I didn't lower this in the first place.
Also note: it needs a documentation update; as we mention the number in the README

Reply:
@chrisohaver : can you update your PR of the README, adding the real change of check for cache. Thong is busy on production issue and will not be available soon to update this PR.

I would propose something like:

```
if ssize < 4 {
		ssize = 4
	}
```

A size of 4 per shard would do a total 2k entries for the smallest cache : I think that is good enough for who want a cache per simple domains. 

Default will remains at 10k cache.

And who want a big cache can still define it.